sdk/file_sdk/Common/std_stream_adapter.cpp

```diff
@@ -0,0 +1,137 @@
+#include <assert.h>
+#include "std_stream_adapter.h"
+
+using std::enable_shared_from_this;
+
+namespace mip {
+namespace file {
+
+StdStreamAdapter::StdStreamAdapter(std::shared_ptr<std::iostream> stdStream)
+  : mInputStream(static_pointer_cast<std::istream>(stdStream)),
+    mOutputStream(static_pointer_cast<std::ostream>(stdStream))
+{}
+
+StdStreamAdapter::StdStreamAdapter(std::shared_ptr<std::ostream> stdOutputStream)
+  : mOutputStream(stdOutputStream)
+{}
+
+StdStreamAdapter::StdStreamAdapter(std::shared_ptr<std::istream> stdInputStream)
+  : mInputStream(stdInputStream)
+{}
+
+StdStreamAdapter::StdStreamAdapter(std::shared_ptr<StdStreamAdapter> from)
+  : mInputStream(from->mInputStream),
+    mOutputStream(from->mOutputStream)
+{}
+
+int64_t StdStreamAdapter::Read(uint8_t *buffer, int64_t bufferLength) {
+  if (mInputStream.get() == nullptr) {
+    throw StdStreamIOException("Operation unavailable");
+  }
+
+  auto fail = mInputStream->fail();
+  // Windows implementation of StringStream has a bug
+  // that for an empty stream fail bit is set after calling seekg(0).
+  // Workaround the bug by calling clear() which clears the failures.
+  if(fail){
+    mInputStream->clear();
+  }
+  mInputStream->read(reinterpret_cast<char *>(buffer), bufferLength);
```

Comment:
>reinterpret_cast [](start = 21, length = 16)

better you use static_cast

Reply:
cannot use static_cast from unsigned to signed char

---
In reply to: [138536249](https://github.com/AzureAD/rms-sdk-for-cpp/pull/151#discussion_r138536249) [](ancestors = 138536249)

Reply:
why is it signed char? shouldn't it be unsigned char?